web/app/themes/clarity/inc/admin/prior-party/prior-party-banner-events.php

```diff
@@ -0,0 +1,215 @@
+    {
+        /**
+         * A post_id was passed, so we only need to get the details for that post.
+         */
+        if ($post_id) {
+            $all_details = get_metadata('post', $post_id, $this->event_details_field);
+            return [$post_id => $this->filterTrackEvents($all_details, $from, $to)];
+        }
+
+        /**
+         * A post_id was not passed, so return results for multiple posts.
+         */
+
+        // The base query arguments.
+        $wp_query_args = [
+            'fields' => 'ids',
+            'posts_per_page'    => -1,
+            'post_type' => ['post', 'page', 'news', 'note-from-antonia'],
+            'post_status' => ['publish', 'pending'],
+            'meta_query' => [
+                'relation' => 'AND',
+                [
+                    'key' => $this->event_timestamp_field,
+                    'compare' => 'EXISTS',
+                ],
+                [
+                    'key' => $this->event_timestamp_field,
+                    'compare' => '!=',
+                    'value' => ''
+                ]
+            ]
+        ];
+
+        // Add clause for date from.
+        if ($from !== null) {
+            $wp_query_args['meta_query'][] = array(
+                'key'     => $this->event_timestamp_field,
+                'value'   => $from,
+                'compare' => '>=',
+                'type'    => 'NUMERIC'
+            );
+        }
+
+        // And, clause for date to.
+        if ($to !== null) {
+            $wp_query_args['meta_query'][] = array(
+                'key'     => $this->event_timestamp_field,
+                'value'   => $to,
+                'compare' => '<=',
+                'type'    => 'NUMERIC'
+            );
+        }
+
+        // Get the posts.
+        $posts = new WP_Query($wp_query_args);
+
+        // Create an array to store the filtered track events.
+        $all_post_events = [];
+
+        foreach ($posts->posts as $post_id) {
+            $all_events = get_metadata('post', $post_id, $this->event_details_field);
+            $all_post_events[$post_id] = $this->filterTrackEvents($all_events, $from, $to);
+        }
+
+        return $all_post_events;
+    }
+
+    /**
+     * Transform the event array into a readable format.
+     *
+     * @param array $event The event.
+     *
+     * @return string
+     */
+    public function eventToReadableFormat(array $event): string
+    {
+        if (empty($event)) {
+            return '';
+        }
+
+        $time = date($this->date_format_time, $event['time']);
+        $user = get_user_by('id', $event['user_id']);
+        $user_name = $user ? $user->display_name : 'Unknown';
+
+        return "User: $user_name,<br/> Action: {$event['action']},<br/> Time: $time";
+    }
+
+    // TODO: lifecycle policy, delete events older than x? Or keep only the most recent x events per post?
```

Comment:
@EarthlingDavey is this TODO for a later date?

Reply:
Cheers, I've got to come back to these.